### PR TITLE
Release signing setup

### PR DESCRIPTION
**Pull request by dominikschulz**
This is adding release signing using the newly created dedicate release signing key 0x21FF4D8F35D1AEE5.
The private key is placed in a secret and [imported](https://github.com/marketplace/actions/import-gpg) into the gpg keyring of the release container.

The `GOPATH` issue is addressed by setting it to a [fixed value](https://laszlo.cloud/setting-gopath-in-github-actions). Let's see how that works. In doubt we might be able to remove trimpath from gcflags and asmflags, but then the path might still show up in some locations.

For a fully working GoReleaser-on-GHA setup at least `GITHUB_TOKEN` is still missing.

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: .github/workflows/autorelease.yml [hunk 1]
```diff
@@ -23,11 +23,33 @@ jobs:
...
+          echo ${GOROOT}
+          env
       -
         name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v2
         with:
           version: latest
           args: release --rm-dist
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (AnomalRoil) [inline .github/workflows/autorelease.yml:33]: The `secrets.GITHUB_TOKEN` is an auto-generated token, no? Why change it?

Comment (AnomalRoil) [reply .github/workflows/autorelease.yml:33]: See https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret

File: internal/updater/verify.go [hunk 2]
```diff
@@ -11,237 +11,23 @@ import (
 	"golang.org/x/crypto/openpgp"
 )
 
+// gpg --expert --full-generate-key
+// (10) ECC (sign only)
+// (5) NIST P-512 (see https://github.com/golang/crypto/blob/eec23a3978ad/openpgp/packet/public_key.go#L227)
+// 2y
```
Comment (AnomalRoil) [inline internal/updater/verify.go:17]: I'm totally not looking forward to the time where we'll have to rotate that key.

File: .goreleaser.yml [hunk 3]
```diff
@@ -133,3 +133,4 @@ signs:
   -
     id: gopass
     artifacts: checksum
+    args: ["-u", "0x21FF4D8F35D1AEE5", "--armor", "--output", "${signature}", "--detach-sign", "${artifact}"]
```
Comment (AnomalRoil) [inline .goreleaser.yml:136]: Maybe use the "--batch" args here? I believe it is passed to `gpg`

**Review by AnomalRoil — APPROVED** [review]

File: .goreleaser.yml (hunk 3 above)
Comment (dominikschulz) [reply .goreleaser.yml:136]: Shouldn't be necessary: https://goreleaser.com/customization/sign/

File: internal/updater/verify.go (hunk 2 above)
Comment (dominikschulz) [reply internal/updater/verify.go:17]: Yes ... but I think we can just add another key to the byte slice and it should just work. If I understood the golang.org/x/crypto/openpgp docs correctly.

File: .github/workflows/autorelease.yml (hunk 1 above)
Comment (dominikschulz) [reply .github/workflows/autorelease.yml:33]: I guess we can re-use that.

**Review by AnomalRoil — APPROVED** [review]

**Pull request merged**